### PR TITLE
Fix 20449 - integer literal specification and implementation differs

### PR DESCRIPTION
**Pull request by dkorpel**
This is assuming we want to adjust the implementation to conform to the spec, and not the other way around.

### CODE REVIEW

File: src/dmd/lexer.d [hunk 1]
```diff
@@ -2107,18 +2107,6 @@ class Lexer
...
-                {
-                    error("signed integer overflow");
-                    err = true;
-                }
-                result = TOK.uns64Literal;
-            }
-            else
-                result = TOK.int64Literal;
-            break;
         case FLAGS.long_:
```
Comment (Geod24) [inline src/dmd/lexer.d]: Just remove this line then ?

**Comment by berni44** [comment]
What about `auto x = 9223372036854775808L;`? Will this compile in the new version?

According to the specs it shouldn't, but it feels a little bit strange to me that adding an L suffix *reduces* the amount of accepted values.

**Comment by Geod24** [comment]
> According to the specs it shouldn't, but it feels a little bit strange to me that adding an L suffix reduces the amount of accepted values.

`L` is asking for a type explicitly. It's a bit redundant with `auto x = long(value);`, but that syntax is only available since a few years while the suffix approach dates back from C.
So if you ask for a type but the argument you provide does not fit in, it makes sense that the compiler errors out, doesn't it ?

**Comment by berni44** [comment]
> `L` is asking for a type explicitly.

Does it? It could also be seen, that this literal uses 64 bits and not only 32.

Anyway, I just wanted to point out, that there might be a problem. For me, the fix is fine.

**Comment by WalterBright** [comment]
Because it's been implemented this way for 19 years now, and the compiler behavior matches C, I recommend that the spec change rather than the compiler.

**Comment by WalterBright** [comment]
I changed the bugzilla issue to be a spec bug rather than a compiler bug.

**Pull request closed without merging**